File: Frontend_EnergyApp/src/app/dtos/UserDto.ts
import {Role} from "./Role";

export class UserDto {
  id: string;
  firstName: string;
  secondName: string;
  email: string;
  role: Role;

  constructor(id: string, firstName: string, secondName: string, email: string, role: Role) {
    this.id = id;
    this.firstName = firstName;
    this.secondName = secondName;
    this.email = email;
    this.role = role;
  }
}

File: UserMicroService/src/main/java/com/userMS/UserMicroService/dtos/userDTOs/UserDTO.java
package com.userMS.UserMicroService.dtos.userDTOs;

import com.userMS.UserMicroService.entities.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserDTO {
    private UUID id;
    private String firstName;
    private String secondName;
    private String email;
    private Role role;
}
